xrpl/utils/xrp_conversions.py

```diff
@@ -0,0 +1,101 @@
+"""Conversions between XRP drops and native number types."""
+
+from decimal import Context, Decimal, InvalidOperation, setcontext
+from re import match
+from typing import Union
+
+from xrpl import XRPLException
+
+ONE_DROP = Decimal("0.000001")  #: Indivisible unit of XRP
+MAX_XRP = Decimal(10 ** 11)  #: 100 billion decimal XRP
+MAX_DROPS = Decimal(10 ** 17)  #: Maximum possible number of drops of XRP
+
+# Drops should be an integer string. MAY have (positive) exponent.
+# See also: https://xrpl.org/currency-formats.html#string-numbers
+_DROPS_REGEX = r"^\s*([1-9][0-9Ee-]{0,17}|0)\s*$"
+
+_DROPS_CONTEXT = Context(prec=18, Emin=0, Emax=18)
+
+
+def xrp_to_drops(xrp: Union[int, float, Decimal]) -> str:
+    """
+    Convert a numeric XRP amount to drops of XRP.
+
+    Args:
+        xrp: Numeric representation of whole XRP
+
+    Returns:
+        Equivalent amount in drops of XRP
+
+    Raises:
+        XRPTypeException: if ``xrp`` is given as a string
+        XRPRangeException: if the given amount of XRP is invalid
+    """
+    if type(xrp) == str:  # type: ignore
+        # This protects people from passing drops to this function and getting
+        # a million times as many drops back.
+        raise XRPTypeException(
```

Comment:
nice!

Reply:
but IMO this should be a vanilla `TypeException`, since that's exactly what it is